### PR TITLE
Api_checkout

### PR DESCRIPTION
**Pull request by kriizjones**
## Pivotal Tracker story 

https://www.pivotaltracker.com/n/projects/1948009

## Changes proposed in this pull request:

This feature does index and delete for checkout page

## What I have learned working on this feature:

• How the database works, what they can do.

• Response Json

• More about pair coding

### CODE REVIEW

**Comment by tochman** [comment]
Any action on this?

**Comment by Philippoes** [comment]
We're adding a sad path after lunch. coverage should go up by then

File: spec/requests/api/v1/checkout_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+RSpec.describe Api::V1::CheckoutController, type: :request do
+    let(:user) { FactoryGirl.create(:user) }
+    let (:dish) { FactoryGirl.create(:dish) }
+    let(:credentials) { user.create_new_auth_token }
+    let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(credentials) }
+
+    describe 'GET /v1/checkout' do
+        it 'should display dishes in order that are not payed' do
+            @order = Order.create(user: user)
```
Comment (tochman) [inline spec/requests/api/v1/checkout_spec.rb]: Use a factory

File: spec/requests/api/v1/checkout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            @order.add(dish, dish.price)
+
+            get '/api/v1/checkout', headers: headers
+            expect(response_json['order_items']).to eq [{ 'name' => 'Pizza', 'price' => 1.0, 'quantity' => 1, 'ready_time' => '10 Jan 16:32' }]
+            expect(response.status).to eq 200
+        end
+
+        it 'send error message if no dishes exist in order' do
+            @order = Order.create(user: user)
```
Comment (tochman) [inline spec/requests/api/v1/checkout_spec.rb]: This is repeated in the block above. Pls extract to a before block but, preferably, use "let" instead

**Review by tochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/checkout_spec.rb (hunk 2 above)
Comment (kriizjones) [reply spec/requests/api/v1/checkout_spec.rb]: Yes 👍

File: spec/requests/api/v1/checkout_spec.rb (hunk 1 above)
Comment (kriizjones) [reply spec/requests/api/v1/checkout_spec.rb]: Done. :)

**Review by tochman — APPROVED** [review]
Good job doing the refactorings!

**Pull request merged**